mantis-runtime-loader/src/main/java/io/mantisrx/runtime/loader/SubscriptionStateHandlerImpl.java

```diff
@@ -53,7 +53,10 @@ class SubscriptionStateHandlerImpl extends AbstractScheduledService implements S
 
     @Override
     public void startUp() {
-        currentState.set(SubscriptionState.of(clock));
+        if (currentState.get() == null) {
```

Comment:
I would convert this into a Precondition instead.

Reply:
I don't think it has to fail if the service got started again. Let me remove the check here directly (atomic set with null should be sufficient.)